File: 卡码/代码随想录/C++/动态规划/dp/05-子序列问题/src/01-子序列(不连续)/LeetCode_0300.cpp
//300. 最长递增子序列
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int lengthOfLIS(vector<int>& nums) {
		vector<int> dp(nums.size(),1);
		int maxLen=0;
		for(int i=1;i<nums.size();i++){
			for(int j=0;j<i;j++){
				if(nums[i]>nums[j]){
					dp[i]=max(dp[i],dp[j]+1);
				}
			}
			maxLen=maxLen>dp[i] ? maxLen : dp[i];
		}
		return maxLen;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/09-动态规划/Java/dp/src/main/java/com/kurumi/dp/_05_子序列问题/_01_不连续子序列/LeetCode_0300.java
//300. 最长递增子序列
package com.kurumi.dp._05_子序列问题._01_不连续子序列;

import java.util.Arrays;

public class LeetCode_0300 {

}
class Solution {
  public int lengthOfLIS(int[] nums) {
    int[] dp=new int[nums.length];
    Arrays.fill(dp, 1);
    int res=1;
    for(int i=1;i<nums.length;i++){
      for(int j=0;j<i;j++)
        if(nums[i]>nums[j]) dp[i]=Math.max(dp[i], dp[j]+1);
      res=res>dp[i] ? res : dp[i];
    }
    return res;
  }
}
